src/main/webapp/user_profile_script.js

```diff
@@ -12,17 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function deleteRow(row) {
-    if (confirm("Are you sure that you want to delete this already published task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("need-help").deleteRow(rowIndex);
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info);
+    return info;
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
+        window.alert("You can only delete an 'OPEN' task.")
+    } else {
+        if (confirm("Are you sure that you want to delete the task?")) {
+            const queryURL = "/tasks?key=" + keyString;
+            const request = new Request(queryURL, {method: "DELETE"});
+            const response = await fetch(request);
+        }
+    }
+}
+
+async function editTask(keyString) {
+    const info = await getInfo(keyString);
+    console.log(info);
+    if (info.status !== "OPEN") {
+        window.alert("You can only edit an 'OPEN' task.")
+    } else {
+        document.getElementById("edit-content-input").value = info.detail;
+        document.getElementById("edit-point-input").value = info.reward.toString();
+        const id_input = document.getElementById("task-id-input");
+        id_input.value = info.keyString;
+        id_input.readOnly = true;
+        id_input.style.display = "none"
```

Comment:
These two properties don't need to be explicitly set if you use `<input type="hidden">`.

Reply:
Okay. After changing the task ID input from `textarea` to hidden input, I removed the two lines in the new commit.